scripts/tt-console/console.c

```diff
@@ -615,13 +615,13 @@ static inline void vuart_putc(struct console *cons, int ch)
 		return;
 	}
 
-	if (tt_vuart_buf_space(vuart->rx_head, vuart->rx_tail, vuart->rx_cap) > 0) {
-		++vuart->rx_tail;
-	}
-
 	volatile char *const rx_buf = (volatile char *)&vuart->buf[vuart->tx_cap];
 
 	rx_buf[vuart->rx_tail % vuart->rx_cap] = ch;
+
+	if (tt_vuart_buf_space(vuart->rx_head, vuart->rx_tail, vuart->rx_cap) > 0) {
+		++vuart->rx_tail;
+	}
```

Comment:
interesting - i can sort of see how this might have been like an off-by-one error.

what kind of error was happening before?